pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -492,6 +489,18 @@ func (r *ReconcileHyperConverged) ensureCDI(instance *hcov1alpha1.HyperConverged
 
 	logger.Info("CDI already exists", "CDI.Namespace", found.Namespace, "CDI.Name", found.Name)
 
+	existingOwners := found.GetOwnerReferences()
+
+	// Previous versions used to have HCO-operator (scope namespace)
+	// as the owner of CDI (scope cluster).
+	// It's not legal, so remove that.
+	if (len(existingOwners) > 0) {
+		logger.Info("CDI has owners, removing...")
+		noOwnerRefs := []metav1.OwnerReference{}
+		found.SetOwnerReferences(noOwnerRefs)
+		r.client.Update(context.TODO(), found)
```

Comment:
Please handle the error.

Reply:
```suggestion
		return r.client.Update(context.TODO(), found)
```

Seems sufficient to me.